GoodMorning/GoodMorning/TODOModalScene/Controllers/TODOModalViewController.swift

```diff
@@ -0,0 +1,207 @@
+        let width = self.view.frame.width * 0.1
+        emojiTextField.snp.makeConstraints { emojiTextField in
+            emojiTextField.width.equalTo(width)
+        }
+
+        let stackView: UIStackView = {
+            let stackView = UIStackView(arrangedSubviews: [emojiTextField, titleTextField])
+            stackView.spacing = 10
+            return stackView
+        }()
+
+        return stackView
+    }
+
+    private func makePriorityLabelView() -> UIView {
+        let label: UILabel = {
+            let label = PretendardLabel(text: "중요도", size: 15, weight: .semibold)
+            label.textColor = .black.withAlphaComponent(0.7)
+            return label
+        }()
+
+        let view: UIView = {
+            let view = UIView()
+            view.addSubview(label)
+            label.snp.makeConstraints { label in
+                label.leading.top.equalToSuperview()
+            }
+
+            view.addSubview(questionButton)
+            questionButton.snp.makeConstraints { questionButton in
+                questionButton.leading.equalTo(label.snp.trailing).offset(4)
+                questionButton.top.equalToSuperview()
+                questionButton.centerY.equalTo(label.snp.centerY)
+            }
+
+            view.addSubview(prioritySegmentedControl)
+            let controlTopOffset: CGFloat = self.view.frame.height * 0.02
+            let controlHeight: CGFloat = self.view.frame.height * 0.047
+            let controlWidth: CGFloat = self.view.frame.width * 0.433
+            prioritySegmentedControl.snp.makeConstraints { control in
+                control.top.equalTo(label.snp.bottom).offset(controlTopOffset)
+                control.leading.equalToSuperview()
+                control.height.equalTo(controlHeight)
+                control.width.equalTo(controlWidth)
+            }
+
+            return view
+        }()
+
+        return view
+    }
+
+    private func setupTextView() {
+        let height = self.view.frame.height * 0.164
+        self.textView.snp.makeConstraints { textView in
+            textView.height.equalTo(height)
+        }
+    }
+
+    @objc private func tappedDeleteButton() {
+        print("delete button 눌림")
+    }
+
+    @objc private func tappedCheckButton() {
+        print("check button 눌림")
+    }
+
+    @objc private func selectedPriority() {
+        self.prioritySegmentedControl.selectedIndex = self.prioritySegmentedControl
+            .selectedSegmentIndex
+    }
+
+}
+
+extension TODOModalViewController: UITextFieldDelegate {
+
+    func textField(
+        _ textField: UITextField,
+        shouldChangeCharactersIn range: NSRange,
+        replacementString string: String
+    ) -> Bool {
+        if string.count == 0 {
+            // 만약 이모티콘이 비었으면 기본값으로 설정.
+            textField.text = "😀"
+            return false
+        }
+
+        if string.count > 0 {
+            textField.text = string
+            return false
+        }
+
+        return true
+    }
+
+}
```

Comment:
이렇게 이모지 키보드를 구현할 수 있다는 점이 정말 신기하네요!!😄 고생하셨습니다!!~
여기서 제가 궁금한 점이 있는데요! 혹시 아래와 같이 구현해도 문제가 없을까요? return true를 탈 경우가 없지 않나해서요! (현재 emoji textfield 말고는 다른 textfield는 해당 delegate 메소드를 활용하지 않으니까요!)

또한, 현재 상태에서는 한글 자판으로 넘어가면 한글을 작성할 수 있더라고요!  그래서 Character extension을 통해서 이모지인 지 확인하는 과정을 가지는 것이 어떨까합니다! ☺️

관련 블로그 글: https://betterprogramming.pub/understanding-swift-strings-characters-and-scalars-a4b82f2d8fde

```suggestion

extension TODOModalViewController: UITextFieldDelegate {

    func textField(
        _ textField: UITextField,
        shouldChangeCharactersIn range: NSRange,
        replacementString string: String
    ) -> Bool {
        guard Character(string).isEmoji else { return false }

        if string.count == 0 {
            // 만약 이모티콘이 비었으면 기본값으로 설정.
            textField.text = "😀"
            return false
        } else {
            textField.text = string
            return false
        }
    }

}

extension Character {

    var isEmoji: Bool {
        guard let firstScalar = unicodeScalars.first else {
            return false
        }
        return firstScalar.properties.isEmoji && firstScalar.value > 0x238C
    }

}
```

Reply:
오! return true를 탈 경우가 없다는 미로의 말에 동의합니다! 그런데 생각해보니 string.count == 0일 때 기본값 이모지를 설정하는 것 보다, string.count == 0일 때 return false로 설정하는 것이 더 좋겠다는 생각이 들었어요. 삭제는 안되고 새로운 값을 눌러야 값이 변경되는 것이죠!

이모지가 아닌 입력값이 들어왔을 때 관련한 구현은 수정 완료했습니다! 스칼라 코드를 사용해 입력값이 이모지인지 아닌지 확인하는 방법이 매우 좋다고 생각해요👍🏻
> 커밋 : 596f69afd0716e67ad8add23159f16a97ad86763